src/main/java/org/takes/rq/RqOnce.java

```diff
@@ -52,16 +47,10 @@ public RqOnce(final Request req) {
         super(
             new RequestOf(
                 new Sticky<>(req::head),
-                new Scalar<InputStream>() {
-                    private final Scalar<String> text = new Sticky<>(
-                        new TextOf(req::body)::asString
-                    );
-
-                    @Override
-                    public InputStream value() throws Exception {
-                        return new InputStreamOf(this.text.value());
-                    }
-                }
+                new Mapped<>(
+                    InputStreamOf::new,
+                    new Sticky<>(new RqPrint(req)::asString)
```

Comment:
@baudoliver7 `RqPrint` is cached. I think we don't need `Sticky` at all here. 
```new RequestOf(new RqPrint(req).head(), new RqPrint().body())```

Reply:
@andreoss Thanks you for your suggestion. The right code is : 
```java
public RqOnce(final Request req) {
    super(
        new RequestOf(
            new RqPrint(req)::head,
            new RqPrint(req)::body
        )
    );
}
```
But unfortunately, It fails tests in `RqOnceTest` because we read twice head and body when calling RqPrint twice.

Reply:
@andreoss The code you suggest raise checked exceptions.

Reply:
@andreoss Is there something else to do ?

Reply:
@andreoss ping

Reply:
@baudoliver7 Nothing else here than, besides the merge conflict
Sorry for the late response.